woocommerce/woocommerce-ios
# Refund Summary → Calculate Total Refunded

Ref #2768. 

This adds calculation for the “Previously Refunded” row in the `RefundConfirmationViewModel`. 

<img src="https://user-images.githubusercontent.com/198826/95793642-47e00b80-0ca3-11eb-8bcd-e670238a8a32.png" width="300">
 
It's not a fake number anymore. 🤸 

## Changes

I added a new `UseCase` for the calculation, [`TotalRefundedCalculationUseCase`](https://github.com/woocommerce/woocommerce-ios/blob/issue/2768-add-refund-total/WooCommerce/Classes/ViewRelated/Orders/Order%20Details/Issue%20Refunds/TotalRefundedCalculationUseCase.swift), and used that in the `ViewModel`. 

Side note: I sometimes wonder if this should be in Yosemite since it's business logic. 🤷 But that requires us to transfer `CurrencySettings` and `CurrencyFormatter` to Yosemite too. 

### Net Calculation

The Net calculation also has the total refunded logic so I replaced that with the `UseCase` as well. 

<img src="https://user-images.githubusercontent.com/198826/95793936-f4ba8880-0ca3-11eb-9662-5d50e2ecdbef.png" width="300">

## Testing

1. Navigate to an order that has more than one refund items. 
2. In the Order Details, confirm that the Net value is still the same as what you would see in `develop`. 
3. Tap on Issue Refund. 
4. Tap Next. 
5. Confirm that the Previously Refunded field shows the sum of all the refund totals. 

## Submitter Checklist
